plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java

```diff
@@ -109,13 +111,44 @@ public HudiTableHandle getTableHandle(ConnectorSession session, SchemaTableName
         }
         return new HudiTableHandle(
                 tableName.getSchemaName(),
-                tableName.getTableName(),
+                name.getTableName(),
                 table.get().getStorage().getLocation(),
-                HoodieTableType.COPY_ON_WRITE,
+                name.getTableType().toString(),
                 TupleDomain.all(),
                 TupleDomain.all());
     }
 
+    @Override
+    public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        return getRawSystemTable(session, tableName)
+                .map(systemTable -> new ClassLoaderSafeSystemTable(systemTable, getClass().getClassLoader()));
+    }
+
+    private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        HudiTableName name = HudiTableName.from(tableName.getTableName());
+        if (name.getTableType() == TableType.COPY_ON_WRITE || name.getTableType() == TableType.MERGE_ON_READ) {
+            return Optional.empty();
+        }
+
+        Optional<Table> tableOptional = metastore.getTable(tableName.getSchemaName(), name.getTableName());
+        if (tableOptional.isEmpty()) {
+            return Optional.empty();
+        }
+        SchemaTableName systemTableName = new SchemaTableName(tableName.getSchemaName(), name.getTableNameWithType());
+        Configuration configuration = hdfsEnvironment.getConfiguration(new HdfsContext(session), new Path(tableOptional.get().getStorage().getLocation()));
```

Comment:
These variables are used only when the table type is timeline. I would move to under `case TIMELINE`.

Reply:
Great, thanks for suggestion